app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt

```diff
@@ -32,12 +30,18 @@ constructor(private val db: FirebaseFirestore, private val auth: FirebaseAuth) :
         .document(userId)
         .get()
         .addOnSuccessListener { document ->
-          val user = document.data?.let { deserializeUser(it) }
-          if (user != null) {
-            onSuccess(user)
-          } else {
-            onFailure(Exception("User not found"))
-          }
+          val userPublic = deserializeUser(document.data!!)
+
+          db.collection(collectionPath)
+              .document(userId)
+              .collection("private")
+              .document("details")
+              .get()
+              .addOnSuccessListener { detailsDocument ->
+                val userDetails = deserializeUserDetails(detailsDocument.data!!)
+                onSuccess(User(userPublic, userDetails))
+              }
+          onSuccess(User(userPublic, null))
```

Comment:
Isn't this line going to always be executed? Since the `onSuccess` call two lines above will be called asynchronously. One solution would be to add this line in the `onFailure` listener of the second database call don't you think?

Reply:
Your implementation sounds better than mine. This way we don't call onSuccess twice